server/src/test/java/io/crate/expression/symbol/LiteralTest.java

```diff
@@ -111,8 +110,14 @@ public void testCompareNestedArrayValues() throws Exception {
     }
 
     @Test
-    public void testCasting() {
+    public void test_cast_int_to_bigint() {
```

Comment:
```suggestion
    public void test_cast_literal_return_function() {
```

Previously this test asserted the already removed special `Literal.cast` logic.
I've kept it to assert the opposite, ensuring that no special literal cast logic exist.
Not sure if we should keep it but I'd suggest a different name expressing the current intention.

Reply:
i'll drop it then if it was an assertion for the special `Literal.cast` logic.

Reply:
I'd prefer to keep it with a concrete name

Reply:
pushed the fixup, i am not sure about the name, i've named it `test_cast_on_literal_is_not_evaluated_immediately`, but maybe smth like test_cast_on_literal_is_not_normalized` would be better?

Reply:
The test ensures that a cast function is returned on casting literals. So the name should express it. After normalization or evaluation the literal will be indeed converted to a new literal, which is not happening here.
Thats why it should be called `test_cast_on_literal_returns_cast_function` or similar.